### PR TITLE
Add observable_filter to general EmulationConfig

### PR DESCRIPTION
**Pull request by raymondEhlers**
This allows for filtering all_observables again based on the overall emulation config

@jdmulligan This should fix the issue you mentioned in #15 . You just need to get the EmulationConfig and pass the corresponding observable_filter in that function

### CODE REVIEW

**Comment by jdmulligan** [comment]
Great, thanks!

File: src/bayesian_inference/mcmc.py [hunk 1]
```diff
@@ -59,23 +59,22 @@ def run_mcmc(config, closure_index=-1):
...
     # In the case of a closure test, we use the pseudodata from the validation design point
-    experimental_results = data_IO.data_array_from_h5(config.output_dir, 'observables.h5', pseudodata_index=closure_index)
+    experimental_results = data_IO.data_array_from_h5(config.output_dir, 'observables.h5', pseudodata_index=closure_index, observable_filter=emulation_config.observable_filter)
 
     # TODO: By default the chain will be stored in memory as a numpy array
     #       If needed we can create a h5py dataset for compression/chunking
 
     # We can use multiprocessing in emcee to parallelize the independent walkers
-    with Pool() as pool:
+    with Pool(processes=1) as pool:
```
Comment (jdmulligan) [inline src/bayesian_inference/mcmc.py:70]: Is there a particular reason for this change? (We still have a lot of optimizing to do for MCMC so totally fine if this is needed for now...but just wanted to check)

Comment (raymondEhlers) [reply src/bayesian_inference/mcmc.py:70]: No - I just used it to simplify debugging and was overzealous with my commit, sorry!

Comment (raymondEhlers) [reply src/bayesian_inference/mcmc.py:70]: Will fix in the morning

Comment (jdmulligan) [reply src/bayesian_inference/mcmc.py:70]: Ah ok :) No worries, thanks! And no rush.

Comment (raymondEhlers) [reply src/bayesian_inference/mcmc.py]: This is now fixed - merging

**Pull request merged**